File: hash/leetcode_3483/solution.js
/**
 * @param {number[]} digits
 * @return {number}
 */
var totalNumbers = function(digits) {
    const uniqueNumbers = new Set();
    const counts = {};
    const evenDigits = [0, 2, 4, 6, 8];

    for(const digit of digits) {
        counts[digit] = (counts[digit] || 0) + 1;
    }

    for(let i=0; i<10; i++) {
        if(!counts[i] || !i) continue;

        counts[i] --;

        for(let j=0; j<10; j++) {
            if(!counts[j]) continue;

            counts[j] --;

            for (let k=0; k<10; k++) {
                if(!counts[k] || !evenDigits.includes(k)) continue;

                const num = i * 100 + j * 10 + k;
                uniqueNumbers.add(num);
            }

            counts[j]++;
        }

        counts[i]++;
    }

    return uniqueNumbers.size;
};

File: hash/leetcode_3483/solution.py
from collections import Counter

class Solution:
    def totalNumbers(self, digits: List[int]) -> int:
        unique_numbers = set()
        counts = Counter(digits)

        for i in range(1, 10):
            if counts[i] == 0:
                continue
            
            counts[i] -= 1

            for j in range(10):
                if counts[j] == 0:
                    continue

                counts[j] -= 1

                for k in range(0, 10, 2):
                    if counts[k] == 0:
                        continue

                    num = i * 100 + j * 10 + k
                    unique_numbers.add(num)

                counts[j] += 1
            counts[i] += 1
        
        return len(unique_numbers)